src/tool/python/code_writers.h

```diff
@@ -72,638 +104,283 @@ namespace xlang
+    void write_type_base(writer& w, TypeDef const& type)
     {
-        struct struct_field_type : public signature_handler_base<struct_field_type>
+        if (has_dealloc(type))
         {
-            std::string field_type{};
-
-            using signature_handler_base<struct_field_type>::handle;
+            w.write("bases.get()");
+        }
+        else
+        {
+            w.write("nullptr");
+        }
+    }
 
-            void handle(GenericTypeInstSig const& type) { handle(get_ireference_type(type)); }
-            void handle_struct(TypeDef const& /*type*/) { field_type = "PyObject*"; }
-            void handle(ElementType type) { field_type = writer::get_cpp_type(type); }
-        };
+    void write_ns_module_exec_init_python_type(writer& w, TypeDef const& type)
+    {
+        if (is_exclusive_to(type)) return;
 
-        struct_field_type sft{};
-        sft.handle(field.Signature().Type());
-        w.write(sft.field_type);
+        w.write("py::winrt_type<%>::python_type = reinterpret_cast<PyTypeObject*>(py::register_python_type(module, _type_name_@, &_type_spec_@, %).detach());\n",
+            bind<write_python_wrapper_template_type>(type),
+            type.TypeName(),
+            type.TypeName(),
+            bind<write_type_base>(type));
     }
 
-    void write_struct_field_initalizer(writer& w, Field const& field)
+    void write_ns_module_exec_func(writer& w, cache::namespace_members const& members)
     {
-        struct struct_field_initializer : public signature_handler_base<struct_field_initializer>
+        w.write("static int module_exec(PyObject* module)\n{\n");
         {
-            using signature_handler_base<struct_field_initializer>::handle;
-
-            struct_field_initializer(writer& wr, std::string_view fn) : w(wr), field_name(fn) {}
-
-            std::string_view field_name;
-            writer& w;
+            writer::indent_guard g{ w };
+            w.write("py::pyobj_handle bases { PyTuple_Pack(1, py::winrt_type<py::winrt_base>::python_type) };\n\n");
+            w.write("try\n{\n");
 
-            void handle_enum(TypeDef const& type)
             {
-                w.write("static_cast<%>(_%)", type, field_name);
-            }
+                writer::indent_guard gg{ w };
 
-            void handle_struct(TypeDef const& type)
-            {
-                w.write("py::converter<%>::convert_to(_%)", type, field_name);
+                settings.filter.bind_each<write_ns_module_exec_init_python_type>(members.classes)(w);
+                settings.filter.bind_each<write_ns_module_exec_init_python_type>(members.interfaces)(w);
+                settings.filter.bind_each<write_ns_module_exec_init_python_type>(members.structs)(w);
             }
 
-            void handle(GenericTypeInstSig const& type)
-            {
-                handle(get_ireference_type(type));
-            }
+            w.write(R"(}
+catch(...)
+{
+    return -1;
```

Comment:
This looks suspicious. Why are we throwing away any trace of the error.

Reply:
yup. Adding py::to_PyErr(); statement